py/nightwatch/plots/camfiber.py

```diff
@@ -291,3 +296,156 @@ def plot_per_fibernum(cds, attribute, cameras, titles={},
+    # Compute the elevation of the moon
+    lst = header["TCSST"].split(":")
+    zenith = radec_to_xyz(15*(float(lst[0])+float(lst[1])/60.0+float(lst[2])/3600.0), float(header["OBS-LAT"]))
+    moon = radec_to_xyz(header["MOONRA"], header["MOONDEC"])
+    return np.arcsin(np.dot(zenith,moon))*180.0/np.pi
+
+def sun_elevation(header):
+    # Compute the elevation of the sun and phase of moon
+    lst = header["TCSST"].split(":")
+    zenith = radec_to_xyz(15*(float(lst[0])+float(lst[1])/60.0+float(lst[2])/3600.0), float(header["OBS-LAT"]))
+    try: sun = radec_to_xyz(header["SUNRA"], header["SUNDEC"])
+    except:
+        # These keywords were only added in mid-February 2021; return False if not available.
+        return False, False
+    moon = radec_to_xyz(header["MOONRA"], header["MOONDEC"])
+    phase = (1.0-np.arccos(np.dot(moon,sun))/np.pi)   # 0 = moon, sun together; 1 = opposed.
+    return np.arcsin(np.dot(zenith,sun))*180.0/np.pi, phase
+
+def plot_fractionalresidual(fiber, header, expos, position=False, plot_height=500, plot_width=500):
+    if position==False: position = expos
+    bfiber = fiber[np.where(fiber['CAM']=='B')]
+    rfiber = fiber[np.where(fiber['CAM']=='R')]
+    zfiber = fiber[np.where(fiber['CAM']=='Z')]
+    bfiber.sort(order="FIBER")
+    rfiber.sort(order="FIBER")
+    zfiber.sort(order="FIBER")
+    #print(file,len(rfiber))
+
+    limit = 0.05
+    bright_neighbor = (bfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(bfiber["MEDIAN_RAW_FLUX"], 1)) \
+        |(bfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(bfiber["MEDIAN_RAW_FLUX"],-1)) \
+        |(rfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(rfiber["MEDIAN_RAW_FLUX"], 1)) \
+        |(rfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(rfiber["MEDIAN_RAW_FLUX"],-1)) \
+        |(zfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(zfiber["MEDIAN_RAW_FLUX"], 1)) \
+        |(zfiber["MEDIAN_RAW_FLUX"]<limit*np.roll(zfiber["MEDIAN_RAW_FLUX"],-1))
+    print("Excluding", np.sum(bright_neighbor), "fibers with bright neighbors.")
+
+    spectrodata = "/global/cfs/cdirs/desi/spectro/data/" # hardcoded
+    desidata = "/global/cfs/cdirs/desi/users/benjikan/nightwatch/" # HARDCODED
+
+    # position = os.path.join('{:08d}'.format(night), '{:08d}'.format(expid))
+
+    file = glob(desidata+position+"/coordinates-*.fits")[0]
+    fits = fitsio.FITS(file)
+    coords = fits["DATA"].read()
+    coords = coords[np.isfinite(coords['FA_FIBER'])]
+    print(file, len(coords))
+    coords.sort(order='FA_FIBER')
+    dx = coords["FIBER_DX"]*1000
+    dy = coords["FIBER_DY"]*1000
+    dr = np.sqrt(dx*dx+dy*dy)
+    goodfiber = ((coords["FLAGS_COR_1"]&4)>0)&(coords["FLAGS_COR_1"]<65535)&(dr<30)&~bright_neighbor
+
+    file = glob(spectrodata+position+"/fiberassign-*")[0]
+    fits = fitsio.FITS(file)
+    fa = fits["FIBERASSIGN"].read()
+    fa.sort(order="FIBER")
+    print(file, len(fa))
+    try:
+        standard = goodfiber&(fa["OBJTYPE"]=='TGT')&((fa['SV1_DESI_TARGET']&0xe00000000)>0)    # Bits 33..35
```

Comment:
`desispec.fluxcalibration.isStdStar(fa)` is the future-proof version of this, i.e. it is supposed to handle SV2_DESI_TARGET, SV3_DESI_TARGET, DESI_TARGET, etc. (and if that breaks, we'll fix it there so that code light nightwatch doesn't also have to change).